### PR TITLE
enhance: Make GridView.padding configurable

### PR DESCRIPTION
**Pull request by yohom**

### CODE REVIEW

**Comment by Fintasys** [comment]
@yohom Can you explain why this would be useful? It only affects the outer padding of the GridView and I can't see much benefit of having it.
Here is an extreme example. 
Screen Shot 2022-03-16 at 0 48 03

**Comment by yohom** [comment]
image
In my case, I want the GridView to have a padding at the bottom to avoid blocking the GridView with a custom delete button.

File: lib/src/config.dart [hunk 1]
```diff
@@ -29,7 +29,8 @@ class Config {
           const TextStyle(fontSize: 20, color: Colors.black26),
       this.tabIndicatorAnimDuration = kTabScrollDuration,
       this.categoryIcons = const CategoryIcons(),
-      this.buttonMode = ButtonMode.MATERIAL});
+      this.buttonMode = ButtonMode.MATERIAL,
+      this.padding = EdgeInsets.zero});
```
Comment (Fintasys) [inline lib/src/config.dart:33]: Can we rename the property to `gridPadding` to give more context and place it below `horizontalSpacing` ?

**Review by Fintasys — COMMENTED** [review]
Thanks for giving more information. 
Theoretically it would be possible for you to make the changes by using `customWidget` and copy the content of `DefaultEmojiPickerView` and make your adjustments. 
But it could be also useful for other people, so if you make the changes I comment above I think we can merge it 👍

File: lib/src/config.dart (hunk 1 above)
Comment (yohom) [reply lib/src/config.dart]: I have modified my code, please check it. 😄

File: lib/src/config.dart [hunk 2]
```diff
@@ -19,6 +19,7 @@ class Config {
       this.emojiSizeMax = 32.0,
       this.verticalSpacing = 0,
       this.horizontalSpacing = 0,
+      this.gridPadding = EdgeInsets.zero,
```
Comment (Fintasys) [inline lib/src/config.dart:22]: Can you also add it to the Readme? Thank you !

**Comment by Fintasys** [comment]
@yohom Sorry that it took so long. Would you able to resolve the conflicts? After that I think it's fine to merge !

**Comment by yohom** [comment]
@Fintasys Hi, conflicts resolved, please check it.

File: README.md [hunk 3]
```diff
@@ -70,20 +71,21 @@ See the [demo](https://github.com/Fintasys/emoji_picker_flutter/blob/master/exam
 | --------------- | ------------------------------------------------------------------ |------------|
 | columns             | Number of emojis per row                                       |7    |
 | emojiSizeMax     | Width and height the emoji will be maximal displayed                 |32.0  |
-| verticalSpacing         | Vertical spacing between emojis | 0    |
+| verticalSpacing         | Verical spacing between emojis | 0    |
```
Comment (Fintasys) [inline README.md:74]: This was corrected in another PR, please revert this change :)

File: README.md [hunk 4]
```diff
@@ -70,20 +71,21 @@ See the [demo](https://github.com/Fintasys/emoji_picker_flutter/blob/master/exam
...
+| gridPadding | The padding of GridView                                                                              | EdgeInsets.zero                                                                                        |
 | initCategory         | The initial Category that will be selected                                                         |Category.RECENT   |
 | bgColor       | The background color of the Widget                                                       |Color(0xFFF2F2F2)    |
 | indicatorColor        | The color of the category indicator                                                       | Colors.blue      |
 | iconColor    | The color of the category icons                                                       | Colors.grey      |
 | iconColorSelected      | The color of the category icon when selected                                                 | Colors.blue |
-| progressIndicatorColor     | The color of the loading indicator during initialization                                | Colors.blue     |
+| progressIndicatorColor     | The color of the loading indicator during initalization                                | Colors.blue     |
```
Comment (Fintasys) [inline README.md:82]: same here

**Comment by yohom** [comment]
@Fintasys Done.
